lib/array.js

```diff
@@ -147,6 +147,27 @@ internals.checkItems = function (items, wasArray, state, options) {
             continue;
         }
 
+        // Ordered
+        if (this._flags.ordered) {
+            res = this._inner.items[v]._validate(item, localState, options);
+            if (!res.errors) {
+                if (this._inner.items[v]._flags.strip) {
+                    internals.fastSplice(items, v);
```

Comment:
Careful with the index here, if your test didn't strip the last one it would fail, you have to change v.